queuelib/queue.py

```diff
@@ -52,13 +54,16 @@ def push(self, string):
         hpos += 1
         szhdr = struct.pack(self.szhdr_format, len(string))
         os.write(self.headf.fileno(), szhdr + string)
-        if hpos == self.chunksize:
+        if hpos == self.chunksize or self.headf.tell() >= self.filesize:  # Roll to next chunk
             hpos = 0
             hnum += 1
             self.headf.close()
             self.headf = self._openchunk(hnum, 'ab+')
         self.info['size'] += 1
         self.info['head'] = [hnum, hpos]
+        
+        if self.syncAll==True:
```

Comment:
:lipstick: 

```suggestion
        if self.syncAll is True:
```

(same below)